llvm/lib/Target/AIE/AIEInterBlockScheduling.cpp

```diff
@@ -136,6 +166,40 @@ checkResourceConflicts(const ResourceScoreboard<FuncUnitWrapper> &Scoreboard,
   return nullptr;
 }
 
+/// Replay the \p SuccBundles top-down into \p ScoreBoard.
+/// If that causes a resource conflict, return an instruction
+/// from \p SuccBundles that is responsible for it.
+/// Note that \p Scoreboard will be modified.
+///
+/// \pre The bundles contain no multi-slot pseudo.
+MachineInstr *
+checkResourceConflictsTopDown(ResourceScoreboard<FuncUnitWrapper> &Scoreboard,
+                              const std::vector<MachineBundle> &SuccBundles,
+                              const AIEHazardRecognizer &HR,
+                              const FixedpointState &Fixedpoint) {
+  DEBUG_LOOPAWARE(dbgs() << "Interblock Predecessor scoreboard:\n";
+                  Scoreboard.dump());
+
+  int TopCycle = 0;
+  MachineInstr *ConflictMI = nullptr;
+  for (const MachineBundle &B : SuccBundles) {
+    for (MachineInstr *MI : B.getInstrs()) {
+      if (HR.getHazardType(Scoreboard, MI->getDesc(), HR.getMemoryBanks(MI),
```

Comment:
Instead of `MI->getDesc()`, please use `*HR.getSelectedAltDescs().getDesc(&MI)`

Reply:
Oh I didn't realise that part was already merged. Will do

Reply:
And that would take away the pre condition of no MSPs?